scripts/beck/st/error_injection.py

```diff
@@ -164,128 +176,156 @@ def ReadMetaData(offset):
 		else:
 			lookupList = random.sample(lookupList, random.randint(1,length))
-			print("Inducing Error at these offsets")
-			print(lookupList)
+			logger.info("Inducing {} Error at these offsets".format(len(lookupList)))
+			logger.info(lookupList)
 			for offset in lookupList:
-				print("**** Inducing Error in Record at offsets {}****".format(hex(offset+8)))
+				logger.info("**** Inducing Error in Record at offsets {}****".format(hex(offset+8)))
 				ReadMetaData(offset+8)
 				EditMetadata(offset+8)
 				ReadMetaData(offset+8)
+				count = count + 1
+	return count
 
 def InduceErrinDMDRecords():
-	print("++++ Induce Random number of errors in All DMD Records ++++")
+	count = 0
+	read_metadata_file()
+	logger.info("++++ Induce Random number of errors in All DMD Records ++++")
 	for recType in DMDList:
-		print("Record Name: {}".format(recType))
+		logger.info("Record Name: {}".format(recType))
 		lookupList = recordDict[recType]
 		length = len(lookupList)
 		if length == 0:
-			print("Record List is empty. Moving to Next Record")
+			logger.info("Record List is empty. Moving to Next Record")
 		else:
 			lookupList = random.sample(lookupList, random.randint(1,length))
-			print("Inducing {} Error at these offsets".format(len(lookupList)))
-			print(lookupList)
+			logger.info("Inducing {} Error at these offsets".format(len(lookupList)))
+			logger.info(lookupList)
 			for offset in lookupList:
-				print("**** Inducing Error in Record at offsets {}****".format(hex(offset+8)))
+				logger.info("**** Inducing Error in Record at offsets {}****".format(hex(offset+8)))
 				ReadMetaData(offset+8)
 				EditMetadata(offset+8)
 				ReadMetaData(offset+8)
+				count = count + 1
+	return count
 
 #Corrupt Metadata file from random location till end of metadata file
 def induceHugeError():
+	count = 0
 	with open(filename, 'r+b') as wbfr:
-		print("** Corrupting 8byte of Metadata at offset {} with b'1111222244443333' **".format(hex(offset)))
+		logger.info("** Corrupting 8byte of Metadata with b'1111222244443333' all place")
 		wbfr.seek(-1, os.SEEK_END)
 		endoffset=wbfr.tell()
-		start = random.randint(1, endoffset)
+		offset = random.randint(1, endoffset)
+		logger.info("Start offset is {}".format(offset))
 		while 1:
-			offset=start+8
+			offset=offset+8
 			wbfr.seek(offset)
 			byte=wbfr.read(8)
 			if not byte:
 				break
 			else:
-				wbfr.seek(offset)
-				wbfr.write(b'\x33\x33\x44\x44\x22\x22\x11\x11')
-		#ReadMetaData(offset)
+				EditMetadata(offset+8)			
+				#wbfr.seek(offset)
+				count = count + 1
+	return count
 
 #Corrupt 512k Metadata in Metadata file from random location
 def induce512kbError():
+	count = 0
 	i = 0
 	j = 0
 	with open(filename, 'r+b') as wbfr:
-		#print("** Corrupting 8byte of Metadata at offset {} with b'1111222244443333' **".format(hex(offset)))
+		#logger.info("** Corrupting 8byte of Metadata at offset {} with b'1111222244443333' **".format(hex(offset)))
 		wbfr.seek(-524400, os.SEEK_END)
```

Comment:
512K == 512 << 10 == 524288

Please explain or add comment to the number "524400".

Reply:
Took a bigger number than 512k so a few records may left without corruption for smooth run.